src/calibrationlib/calibration_table_generator.cpp

```diff
@@ -0,0 +1,47 @@
+#include "calibration_table_generator.hpp"
+#include "storage/table.hpp"
+
+namespace opossum {
+
+CalibrationTableGenerator::CalibrationTableGenerator(std::shared_ptr<TableGeneratorConfig> config) {
+  // Generate all possible permutations of column types
+  for (DataType data_type : config->data_types) {
+    for (EncodingType encoding_type : config->encoding_types) {
+      if (encoding_supports_data_type(encoding_type, data_type)) {
+        for (ColumnDataDistribution column_data_distribution : config->column_data_distributions) {
```

Comment:
auto

Reply:
See above. I changed this to const auto.